google/ggrc-core
# GGRC-8834 / GGRC-8828 Fix revision's created date for Scope objects

# Issue description

When performing some actions on Scope objects on the GGRC side (e.g. attaching GDrive folder), new object's revisions are displayed in change log with a wrong date value.

The reason is that revisions of `Synchronizable` objects had `created_at` and `updated_at` being calculated as object's `updated_at` date. And this should be so only in case an action on an objects is being performed by external user.

# Steps to test the changes

1. Attach GDrive folder to any Scope object;
2. Open "Change Log" tab;
3. Make sure that current date is shown for the record about attached GDrive folder.

# Solution description

* Add additional check for external user during revision's `created_at` and `updated_at` population.

* Refactor `_revision_generator()` by moving field population logic into separate functions.

# Sanity checklist




# PR Review checklist
